docs/SemanticARC.md

```diff
@@ -68,6 +68,13 @@ We now go into depth on each one of those points.
 
 The first step towards implementing Semantic ARC is to split the "Canonical SIL Stage" into two different stages: High Level and Low Level SIL. The main distinction in between the two stages is, that in High Level SIL, ARC semantic invariants will be enforced via extra conditions on the IR. In contrast, once Low Level SIL has been reached, no ARC semantic invariants are enforced and only very conservative ARC optimization may occur. The intention is that Low Level SIL would /only/ be used when compiling with optimization enabled, so both High and Low Level SIL will necessarily need to be able to be lowered to LLVM IR.
 
+[EE]
+It sounds like that the high level SIL is lowered to the low level SIL by replacing the new high level ARC operations by something similar to what we have now, e.g. retain/release instructions.
+If my understanding is correct, I have this comment:
+I think this is not really needed. Instead it's sufficient to run one (or few) "ARC-insertion" pass(es) at the very end of the pipeline, which determines where retains/releases should be generated in IRGen. These passes would be the combination of our current ARCSequenceOpt and ARC code motion passes.
+These "low-level" passes still could insert retain/release instructions, but those instructions could be just viewed as "markers" for IRGen to specify where ARC operations should be generated, and they would not have any further semantic meaning in SIL.
+So in other words: we would not give up the semantic ARC rules at any time in the pipeline. We would just insert ARC "marker" instructions at the end of the pipeline, which are used by IRGen.
+
```

Comment:
I reflected on this. My main thoughts here were that this would be a staging thing. I should have made that clear and moved it to the implementation section.